### DISCUSSION TITLE
Sending variable into functions without them showing up as an option in slash commands

### DISCUSSION THREAD

**Discussion by TheKrafter** [nextcord/nextcord#963] [Q&A] [answered]
I want to send a global variable into the namespace for the function, but not show it as an option for users. Here's a snippet of a subcommand:
```python
async def add(self, interaction: nextcord.Interaction, user: nextcord.Member, cursor=cursor):
```
`cursor` is the mysql cursor, and calling it directly within the function raised a not defined exception. 

How can i make it so that the `cursor` option doesn't show up, but the `cursor` global variable is still passed into the function definition?

**Comment by ooliver1** [comment, marked as answer]
You could hold an attribute in your bot instance, so it can be accessed from `self` (assuming you set that attribute in your cog [like the examples](https://github.com/nextcord/nextcord/blob/master/examples/application_commands/cog_example.py#L9)), or [`Interaction.client`](https://docs.nextcord.dev/en/stable/api.html#nextcord.Interaction.client). I would recommend storing the connection, and making a new cursor *per* transaction/operation too.

**Reply by TheKrafter** [reply]
tysm
